react-hot-button-component/src/hot-button.jsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+let clickCount = 0;
+
+class CustomButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { isClicked: false };
+  }
+
+  handleClick(props) {
+    this.setState({ isClicked: true });
+  }
+
+  render() {
+    clickCount++;
+    if (clickCount < 4) {
+      return (
+        <div>
+          <button onClick={this.handleClick.bind(this)} className="cold">Hot Button</button>
+        </div>
+      );
+    } else if (clickCount < 7) {
+      return (
+        <div>
+          <button onClick={this.handleClick.bind(this)} className="cool">Hot Button</button>
+        </div>
+      );
+    } else if (clickCount < 10) {
+      return (
+        <div>
+          <button onClick={this.handleClick.bind(this)} className="luke-warm">Hot Button</button>
+        </div>
+      );
+    } else if (clickCount < 13) {
+      return (
+        <div>
+          <button onClick={this.handleClick.bind(this)} className="warm">Hot Button</button>
+        </div>
+      );
+    } else if (clickCount < 16) {
+      return (
+        <div>
+          <button onClick={this.handleClick.bind(this)} className="hot">Hot Button</button>
+        </div>
+      );
+    } else {
+      return (
+        <div>
+          <button onClick={this.handleClick.bind(this)} className="super-hot">Hot Button</button>
+        </div>
+      );
+    }
+  }
```

Comment:
In `render` you can use `this.state.count` to determine which class should be applied to the button, then return a single button at the end.
```JAVASCRIPT
render() {
  const count = this.state.count;
  let btnClass;

  if (count < 4) {
    btnClass = 'cold';
  } else if (count < 7) {
    btnClass = 'cool';
  } else if (count < 10) {
    btnClass = 'luke-warm';
  } else if (count < 13) {
    btnClass = 'warm';
  } else if (count < 16) {
    btnClass = 'hot';
  } else {
    btnClass = 'super-hot';
  }

  return (
    <div>
      <button onClick={this.handleClick.bind(this)} className={btnClass}>Hot Button</button>
    </div>
  );
}
```